File: 1/1000s/13s/1394/FindLuckyIntegerInAnArray.java
/*
https://leetcode.com/problems/find-lucky-integer-in-an-array/description/?envType=daily-question&envId=2025-07-05
*/

class Solution {

    public static boolean isLucky(int[] arr, int val){
        int freq=0;
        for(int num: arr){
            if(num==val) freq++;
        }
        return freq==val;
    }

    public int findLucky(int[] arr) {
        int lengthOfArr=arr.length;
        TreeSet<Integer>allLuckyNumbers=new TreeSet<>();
        for(int i=0; i<lengthOfArr; ++i){
            if(isLucky(arr, arr[i])){
                allLuckyNumbers.add(arr[i]);
            }
        }
        if(allLuckyNumbers.size()==0) return -1;
        return allLuckyNumbers.last();
    }
}

File: 1/1000s/13s/1394/FindLuckyIntegerInAnArray.py
# https://leetcode.com/problems/find-lucky-integer-in-an-array/description/?envType=daily-question&envId=2025-07-05

class Solution:
    def findLucky(self, arr: List[int]) -> int:

        def isLucky(freqMap, val):
            if freqMap[val]==val:
                return True
            return False

        lengthOfArr=len(arr)
        freqMap={}
        for i in range(0, lengthOfArr, 1):
            if arr[i] in freqMap:
                freqMap[arr[i]]=freqMap[arr[i]]+1
            else:
                freqMap[arr[i]]=1
        maxLuckyNumber=-1
        for i in range(0, lengthOfArr, 1):
            if isLucky(freqMap, arr[i]):
                maxLuckyNumber=max(maxLuckyNumber, arr[i])
        return maxLuckyNumber